client/state/initial-state.js

```diff
@@ -235,16 +235,18 @@ async function getInitialStoredState( initialReducer ) {
 	}
 
 	let initialStoredState = await getStateFromLocalStorage( initialReducer );
-	if ( ! initialStoredState ) {
-		return null;
-	}
-
 	const storageKeys = [ ...initialReducer.getStorageKeys() ];
 
 	async function loadReducerState( { storageKey, reducer } ) {
-		const storedState = await getStateFromLocalStorage( reducer, storageKey );
+		let storedState = await getStateFromLocalStorage( reducer, storageKey, false );
+
+		if ( ! storedState && storageKey === 'signup' ) {
+			storedState = await getStateFromLocalStorage( reducer, storageKey, true );
+			debug( 'fetched signup state from logged out state', storedState );
+		}
+
 		if ( storedState ) {
-			initialStoredState = initialReducer( initialStoredState, {
+			initialStoredState = initialReducer( initialStoredState || {}, {
```

Comment:
Now when you taught `applyStoredState` to handle empty `state` with the `get( ... )` call, you don't need this default any more. Passing the `null` state will work.